controllers/articlesController.js

```diff
@@ -156,4 +159,84 @@ export default class ArticlesController {
       return Response.send(res, STATUS.BAD_REQUEST, error, false);
     }
   }
+
+  /**
+ * @description users can bookmark articles for reading later.
+ * @static
+ * @param  {object} req - The request object
+ * @param  {object} res - The response object
+ * @return {object} - It returns the request response object
+ */
+  static async bookmark(req, res) {
+    const { slug } = req.params;
+    const userId = req.user.id;
+    let articleId;
+
+    try {
+      const user = await User.findOne({
```

Comment:
@fejiroofficial Starting from line 176 to 189, kindly leverage on an existing middleware for authenticating users and write a separate middleware for article validation.

Reply:
done